### PR TITLE
Ignore only empty drafts while sending with clear_draft

### PR DESCRIPTION
**Pull request by udf**
Currently tdesktop ignores ALL draft updates for `kSkipCloudDraftsFor` seconds (which is 3 seconds) [372cf275e004c9040d493207eb46ec6e379e9f8f].

This commit makes it only ignore EMPTY drafts (with no message and no reply id) during that timeout.

This behaviour makes it work as good as before without the drawbacks of ignoring actual drafts made within the timeout period.

### CODE REVIEW

File: Telegram/SourceFiles/history/history.cpp [hunk 1]
```diff
@@ -410,8 +410,11 @@ Data::Draft *History::createCloudDraft(Data::Draft *fromDraft) {
 	return cloudDraft();
 }
 
-bool History::skipCloudDraft(const QString &text, TimeId date) const {
-	if (date > 0 && date <= _lastSentDraftTime + kSkipCloudDraftsFor) {
+bool History::skipCloudDraft(const QString &text, MsgId replyTo, TimeId date) const {
+	if (Data::draftStringIsEmpty(text)
+		&& !replyTo
```
[the remark this answers is not in the document]
Comment (udf) [reply Telegram/SourceFiles/history/history.cpp]: When you think about it in terms of "If there isn't a reply vs. If reply is zero", the former makes more sense.

**Comment by john-preston** [comment]
@udf What is the use case of sending a message from desktop and then wanting it to pick up a draft from another device within three seconds?

**Comment by Lonami** [comment]
Userbots. They can enhance the user experience for those who know how to use them and does no harm to those who don't. Any reason *against* the change? I think it's fine to merge this PR.

**Comment by udf** [comment]
@john-preston Like @Lonami said, it's for a userbot that I'm making which tries to make more client features accessible without having to use a mouse. Specifically, it's so I can use tdesktop's ability to reply to a message with ctrl+up to edit a message.

vid here: https://giant.gfycat.com/SorrowfulTepidAsianporcupine.webm

Basically I reply with .e to a message and it makes a draft of that message with "\n.e" appended (and is a reply to the message I want to edit).

It's a rather specific use case, but it doesn't hurt anyone, and I'm sure someone else can come up with another way to use this.

**Comment by JuanPotato** [comment]
+1 because sometimes I'll be typing a message, send, immediately start typing again but it gets cut off and cleared.

**Comment by john-preston** [comment]
@udf Sorry, there are conflicts with the updated code. Please resolve the conflicts so it could be merged in the next version.

**Comment by udf** [comment]
@john-preston Okay, done.

**Comment by udf** [comment]
I've amended a mistake that I made when I was merging, it should build now.

**Comment by john-preston** [comment]
Ok, thanks.

**Comment by john-preston** [comment]
OH SHI~. This PR rolled back libtgvoip 🙈

**Comment by udf** [comment]
I'm bad at git lol. Rebasing would've been better than a merge, but I didnt't know that at the time.

**Pull request merged**